cylc/flow/task_events_mgr.py

```diff
@@ -1293,3 +1317,13 @@ def unset_waiting_event_timer(self, id_key):
         """
         self._event_timers[id_key].unset_waiting()
         self.event_timers_updated = True
+
+    def reset_bad_hosts(self):
+        """Periodically clear bad_hosts list.
```

Comment:
The periodic bit is the responsibility of the caller.
```suggestion
        """Clear bad_hosts list.
```